libethereum/ExtVM.cpp

```diff
@@ -134,3 +134,26 @@ void ExtVM::suicide(Address _a)
 	m_s.subBalance(myAddress, m_s.balance(myAddress));
 	ExtVMFace::suicide(_a);
 }
+
+h256 ExtVM::blockHash(u256 _number)
+{
+	u256 const currentNumber = envInfo().number();
+
+	if (_number >= currentNumber || _number < (std::max<u256>(256, currentNumber) - 256))
+		return h256();
+
+	if (currentNumber < m_sealEngine.chainParams().u256Param("metropolisForkBlock") + 256)
+	{
+		assert(envInfo().lastHashes().size() > (unsigned)(currentNumber - 1 - _number));
```

Comment:
Also `currentNumber` is never 0 here, because you can't execute anything in the genesis block